recording-daemon/packet.c

```diff
@@ -290,12 +284,13 @@ static ssrc_t *ssrc_get(stream_t *stream, unsigned long ssrc) {
 				goto tls_out;
 			}
 			ret->tls_fwd_stream = streambuf_new_ptr(&ret->tls_fwd_poller, ret->ssl, &ssrc_tls_funcs);
+			ssrc_tls_state(ret);
 		} else {
 			ret->tls_fwd_stream = streambuf_new(&ret->tls_fwd_poller, ret->tls_fwd_sock.fd);
+			ret->tls_fwd_poller.state = PS_OPEN;
```

Comment:
I think that's a bit too aggressive now 😆

`connect_socket_nb()` doesn't necessarily give you an open socket immediately, so state should still go to PS_CONNECTING first. But in `ssrc_tls_state()`, with state `PS_CONNECTING`, when `connect_socket_retry()` then returns success, state can be set to `PS_OPEN` instead of `PS_HANDSHAKE`.